packages/client/src/profile/profileReducer.ts

```diff
@@ -122,8 +117,19 @@ export const profileReducer: LoopReducer<
       const result: ApolloQueryResult<GQLQuery> = action.payload
       const data: GQLQuery = result && result.data
 
+      const language = getSelectedLanguage()
+      // Remove token and language from url if these exists
+      if (window.location.search.includes('token=')) {
+        window.history.replaceState(null, '', window.location.pathname)
+      }
+
       if (data && data.getUser) {
         const userDetails = getUserDetails(data.getUser)
+
+        if (language) {
+          userDetails.language = language
```

Comment:
I almost would like to suggest you wouldn't mutate the object, but this reducer changes `userDetails` anyways, so I don't think it makes much of a difference logically. I've just recently seen quite a many mutations all over the app in places where they don't belong, and I think if someone uses this as an example to be used somewhere else 😄  what do you think?